Add LDAP lookup config examples (#20179)
dotnet/AspNetCore.Docs: aspnetcore/security/authentication/windowsauth.md

> Credentials can be persisted across requests on a connection. *Negotiate authentication must not be used with proxies unless the proxy maintains a 1:1 connection affinity (a persistent connection) with Kestrel.*

> [!NOTE]

> > The Negotiate handler detects if the underlying server supports Windows Authentication natively and if it is enabled. If the server supports Windows Authentication but it is disabled, an error is thrown asking you to enable the server implementation. When Windows Authentication is enabled in the server, the Negotiate handler transparently forwards authentication requests to it.

Reviewer: The style guide prefers `it's` over `it is`
I think it reads better with `it's` but it's a trivial nit :)

Author: Bah. You americans and your contractions.

For more information on middleware, see <xref:fundamentals/middleware/index>.

> Kerberos authentication does not provide any role information for an authenticated users. In order to add role and group information to a kerberos user the kerberos  authentication handler must be configured to retreive the roles from an LDAP domain. The most basic configuration only specifies an LDAP domain to query against and will use the authenticated user's context to query the LDAP domain;

Reviewer 2: Even though this API would "work" on Windows, it's mostly only relevant to Linux scenarios.

Author: It's implied further up, @scottaddie should it get repeated down here? It's your doc really

> Kerberos authentication does not provide any role information for an authenticated users. In order to add role and group information to a kerberos user the kerberos  authentication handler must be configured to retreive the roles from an LDAP domain. The most basic configuration only specifies an LDAP domain to query against and will use the authenticated user's context to query the LDAP domain;

Reviewer: ```suggestion
Kerberos authentication doesn't provide any role information for an authenticated users. In order to add role and group information to a kerberos user, the kerberos authentication handler must be configured to retrieve the roles from an LDAP domain. The most basic configuration only specifies an LDAP domain to query against and will use the authenticated user's context to query the LDAP domain;
```

Reviewer 3: ```suggestion
Kerberos authentication on Linux or macOS doesn't provide any role information for an authenticated user. To add role and group information to a Kerberos user, the authentication handler must be configured to retrieve the roles from an LDAP domain. The most basic configuration only specifies an LDAP domain to query against and will use the authenticated user's context to query the LDAP domain:
```

```csharp
services.AddAuthentication(NegotiateDefaults.AuthenticationScheme)
    .AddNegotiate(options =>
    {
        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
        {
            options.EnableLdap("contoso.com");
        }
    });
```

> Some configurations may require specific credentials to query the LDAP domain, this can also be specified in the options;

Reviewer: ```suggestion
Some configurations may require specific credentials to query the LDAP domain, this can also be specified in the options:
```

```csharp
services.AddAuthentication(NegotiateDefaults.AuthenticationScheme)
    .AddNegotiate(options =>
    {
        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
        {
            options.EnableLdap("contoso.com");
            options.MachineAccountName = "machineName";
            options.MachineAccountPassword = "PassW0rd";            
        }
    });
```

> By default the negotiate authentication handler will resolve nested domains, however in a large or complicated LDAP environment this may result in a slow lookup, or a lot of memory being used for each user. Nested domain resolution can be disabled using the `IgnoreNestedGroups` option.

Reviewer: ```suggestion
By default the negotiate authentication handler resolves nested domains. In a large or complicated LDAP environment,  resolving nested domains may result in a slow lookup, or a lot of memory being used for each user. Nested domain resolution can be disabled using the `IgnoreNestedGroups` option.
```
